rasa/importers/autoconfig.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import os
+from pathlib import Path
+
+from typing import Text, Dict, Any, List
+
+from rasa.cli import utils as cli_utils
+from rasa.constants import CONFIG_AUTOCONFIGURABLE_KEYS
+from rasa.utils import io as io_utils
+
+logger = logging.getLogger(__name__)
+
+
+# In future iterations, this function and the function `create_config_for_keys` below
+# will get additional input parameters to allow for smarter configuration. E.g.
+# training_data: TrainingData
+# domain: Domain
+# stories: StoryGraph
+def get_auto_configuration(config_file) -> Dict[Text, Any]:
+    """Determine configuration from a configuration file.
+
+    Keys that are provided in the file are kept. Keys that are not provided are
+    configured automatically.
+    """
+    if config_file and os.path.exists(config_file):
+        config = io_utils.read_config_file(config_file)
+
+        missing_keys = [k for k in CONFIG_AUTOCONFIGURABLE_KEYS if not config.get(k)]
+        create_config_for_keys(config, missing_keys)
+
+        dump_config(config, config_file)
+    else:
+        config = {}
+
+    return config
+
+
+def create_config_for_keys(config: Dict[Text, Any], keys: List[Text]) -> None:
```

Comment:
I think we should aim for pure functions wherever possible - I don't think there is a need for side-effect here and we can as easily copy the dict and return the modified copy